src/db/mod.rs

```diff
@@ -59,10 +61,11 @@ pub const BATCH_LIFETIME: i64 = 2 * 60 * 60 * 1000; // 2 hours, in milliseconds
 /// DbPools' worker ThreadPool size
 pub const DB_THREAD_POOL_SIZE: usize = 50;
 
-type DbFuture<T> = LocalBoxFuture<'static, Result<T, ApiError>>;
+type DbFuture<'a, T> = LocalBoxFuture<'a, Result<T, ApiError>>;
 
+#[async_trait(?Send)]
 pub trait DbPool: Sync + Send + Debug {
```

Comment:
We can revert this back to `#[async_trait]` (adds back Send to the returned future)